source/common/tracing/zipkin/tracer.cc

```diff
@@ -44,16 +50,16 @@ SpanPtr Tracer::startSpan(const std::string& span_name, SystemTime timestamp) {
   return span_ptr;
 }
 
-SpanPtr Tracer::startSpan(const std::string& span_name, SystemTime timestamp,
-                          SpanContext& previous_context) {
+SpanPtr Tracer::startSpan(const Tracing::Config& config, const std::string& span_name,
+                          SystemTime timestamp, SpanContext& previous_context) {
   SpanPtr span_ptr(new Span());
   Annotation annotation;
   uint64_t timestamp_micro;
 
   timestamp_micro =
       std::chrono::duration_cast<std::chrono::microseconds>(timestamp.time_since_epoch()).count();
 
-  if (previous_context.annotationSet().sr_ && !previous_context.annotationSet().cs_) {
```

Comment:
Is it safe to disregard altogether the previous context when making this decision?

Reply:
In normal application tracing, the previous (parent) span kind is not taken into account. Spans are created based on reflecting what they represent - i.e. either a client or server span. In the case of Envoy, what dictates this is whether the communication represents ingress or egress.

Reply:
Agreed! This makes sense.